axelrod/strategies/ann.py

```diff
@@ -0,0 +1,222 @@
+
+        # Calculate output value
+        output_value = 0
+        for i in range(self.hidden_layer_size):
+            output_value += hidden_values[i] * \
+                            self.hidden_to_output_layer_weights[i]
+
+        return output_value
+
+    @init_args
+    def __init__(
+        self,
+        input_to_hidden_layer_weights=[],
+        hidden_to_output_layer_weights=[],
+        bias_weights=[]
+    ):
+
+        Player.__init__(self)
+        self.input_to_hidden_layer_weights = input_to_hidden_layer_weights
+        self.hidden_to_output_layer_weights = hidden_to_output_layer_weights
+        self.bias_weights = bias_weights
+
+        self.input_values = len(input_to_hidden_layer_weights[0])
+        self.hidden_layer_size = len(hidden_to_output_layer_weights)
+
+    def strategy(self, opponent):
+        # Compute features for Neural Network
+        # These are True/False 0/1
+        if len(opponent.history) == 0:
+            opponent_first_c = 0
+            opponent_first_d = 0
+            opponent_second_c = 0
+            opponent_second_d = 0
+            my_previous_c = 0
+            my_previous_d = 0
+            my_previous2_c = 0
+            my_previous2_d = 0
+            opponent_previous_c = 0
+            opponent_previous_d = 0
+            opponent_previous2_c = 0
+            opponent_previous2_d = 0
+
+        elif len(opponent.history) == 1:
+            opponent_first_c = 1 if opponent.history[0] == C else 0
+            opponent_first_d = 1 if opponent.history[0] == D else 0
+            opponent_second_c = 0
+            opponent_second_d = 0
+            my_previous_c = 1 if self.history[-1] == C else 0
+            my_previous_d = 0 if self.history[-1] == D else 0
+            my_previous2_c = 0
+            my_previous2_d = 0
+            opponent_previous_c = 1 if opponent.history[-1] == C else 0
+            opponent_previous_d = 1 if opponent.history[-1] == D else 0
+            opponent_previous2_c = 0
+            opponent_previous2_d = 0
+
+        else:
+            opponent_first_c = 1 if opponent.history[0] == C else 0
+            opponent_first_d = 1 if opponent.history[0] == D else 0
+            opponent_second_c = 1 if opponent.history[1] == C else 0
+            opponent_second_d = 1 if opponent.history[1] == D else 0
+            my_previous_c = 1 if self.history[-1] == C else 0
+            my_previous_d = 0 if self.history[-1] == D else 0
+            my_previous2_c = 1 if self.history[-2] == C else 0
+            my_previous2_d = 1 if self.history[-2] == D else 0
+            opponent_previous_c = 1 if opponent.history[-1] == C else 0
+            opponent_previous_d = 1 if opponent.history[-1] == D else 0
+            opponent_previous2_c = 1 if opponent.history[-2] == C else 0
+            opponent_previous2_d = 1 if opponent.history[-2] == D else 0
+
+        # Remaining Features
+        turns_remaining = self.match_attributes['length'] - len(self.history)
```

Comment:
Can we add a test for the case of a match when the match length is unknown? (Just to make sure this doesn't fall over).